packages/node-opentelemetry/src/otel.ts

```diff
@@ -32,37 +39,46 @@ export type SDKConfig = {
 
 export const setTraceAttributes = hyperDXGlobalContext.setTraceAttributes;
 
+const setOtelEnvs = () => {
+  // set default otel env vars
+  env.OTEL_NODE_RESOURCE_DETECTORS = env.OTEL_NODE_RESOURCE_DETECTORS ?? 'all';
+  env.OTEL_TRACES_SAMPLER = DEFAULT_OTEL_TRACES_SAMPLER;
+  env.OTEL_TRACES_SAMPLER_ARG = DEFAULT_OTEL_TRACES_SAMPLER_ARG;
+};
+
 let sdk: NodeSDK;
 let hdxConsoleInstrumentation: HyperDXConsoleInstrumentation;
 
 export const initSDK = (config: SDKConfig) => {
-  // enable otel debug mode if HDX_DEBUG_MODE_ENABLED is set
-  if (HDX_DEBUG_MODE_ENABLED) {
-    env.OTEL_LOG_LEVEL = 'debug';
-  }
+  hdx('Setting otel envs');
+  setOtelEnvs();
 
-  // set default otel env vars
-  env.OTEL_EXPORTER_OTLP_ENDPOINT =
-    env.OTEL_EXPORTER_OTLP_ENDPOINT ?? 'https://in-otel.hyperdx.io';
-  env.OTEL_NODE_RESOURCE_DETECTORS = env.OTEL_NODE_RESOURCE_DETECTORS ?? 'all';
-  env.OTEL_LOG_LEVEL = env.OTEL_LOG_LEVEL ?? 'none'; // silence by default
-  env.OTEL_TRACES_SAMPLER = env.OTEL_TRACES_SAMPLER ?? 'parentbased_always_on';
-  env.OTEL_TRACES_SAMPLER_ARG = env.OTEL_TRACES_SAMPLER_ARG ?? '1';
+  const stopOnTerminationSignals = config.stopOnTerminationSignals ?? true; // Stop by default
 
-  // patch OTEL_EXPORTER_OTLP_HEADERS to include API key
+  let exporterHeaders;
   if (env.HYPERDX_API_KEY) {
-    env.OTEL_EXPORTER_OTLP_HEADERS = `${env.OTEL_EXPORTER_OTLP_HEADERS},authorization=${env.HYPERDX_API_KEY}`;
+    exporterHeaders = {
+      Authorization: env.HYPERDX_API_KEY,
+    };
   } else {
     console.warn(`${LOG_PREFIX} HYPERDX_API_KEY is not set`);
   }
 
-  const stopOnTerminationSignals = config.stopOnTerminationSignals ?? true; // Stop by default
-
   hdx('Initializing OpenTelemetry SDK');
-  const consoleInstrumentationEnabled = config.consoleCapture ?? true;
+  let consoleInstrumentationEnabled = config.consoleCapture ?? true;
+  if (DEFAULT_OTEL_LOG_LEVEL === DiagLogLevel.DEBUG) {
+    // FIXME: better to disable console instrumentation if otel log is enabled
```

Comment:
fine to leave until later?

Reply:
yeah its fine. currently we disable console instrumentation only if otel level is debug (which caused a recursion issue)